plugin/src/main/kotlin/com/automattic/android/PublishToS3Plugin.kt

```diff
@@ -1,37 +1,48 @@
 package com.automattic.android.publish
 
+import com.android.build.gradle.LibraryExtension
 import org.gradle.api.Project
 import org.gradle.api.Plugin
 import org.gradle.api.publish.maven.tasks.GenerateMavenPom
 import org.gradle.api.publish.maven.tasks.PublishToMavenRepository
-import org.gradle.jvm.tasks.Jar
-import com.android.build.gradle.LibraryExtension
 
 class PublishToS3Plugin : Plugin<Project> {
     override fun apply(project: Project) {
-        project.plugins.apply("maven-publish")
-        project.addS3MavenRepository()
+        applyInternal(project)
+    }
+}
+
+class PublishToS3WithoutSourcesPlugin : Plugin<Project> {
+    override fun apply(project: Project) {
+        applyInternal(project, withSources = false)
+    }
+}
 
-        project.tasks.register("calculateVersionName", CalculateVersionNameTask::class.java)
-        project.tasks.register("isVersionPublishedToS3", CheckS3VersionTask::class.java)
+private fun applyInternal(project: Project, withSources: Boolean = true) {
+    project.plugins.apply("maven-publish")
+    project.addS3MavenRepository()
 
-        project.pluginManager.withPlugin("com.android.library") {
-            project.extensions.findByType(LibraryExtension::class.java)?.let { androidLibrary ->
-                androidLibrary.publishing.singleVariant("release") {
+    project.tasks.register("calculateVersionName", CalculateVersionNameTask::class.java)
+    project.tasks.register("isVersionPublishedToS3", CheckS3VersionTask::class.java)
+
+    project.pluginManager.withPlugin("com.android.library") {
+        project.extensions.findByType(LibraryExtension::class.java)?.let { androidLibrary ->
+            androidLibrary.publishing.singleVariant("release") {
+                if (withSources) {
```

Comment:
Minor (🔍): This can become a one-liner, just to avoid the extra braces and make this look closer to `withJavadocJar()`: `if (withSources) withSourcesJar()`

PS: Feel free to totally ignore.

Reply:
Addressed in 1882483472294e2c2eb71d76efe09e18719d06c5 together with the suggestion above.

Reply:
Thanks @oguzkocer ! 👍